docs/learn.md

```diff
@@ -721,20 +723,1250 @@ template](https://github.com/melange-re/melange-opam-template).
+  properties dynamically by using the bracket notation `[]`
+- [`bs.module`](#using-functions-from-other-javascript-modules): bind to a value
+  from a JavaScript module
+- [`bs.new`](#javascript-classes): bind to a JavaScript constructor
+- [`bs.obj`](#using-jst-objects): create JavaScript object
+- [`bs.return`](#wrapping-returned-nullable-values): automate conversion from
+  nullable values to `Option.t` values
+- [`bs.send`](#calling-a-function-as-an-object-property): call a function that
+  is a JavaScript object property
+- [`bs.set`](#bind-to-object-properties): set JavaScript object properties
+  statically by name
+- [`bs.set_index`](#bind-to-object-properties): set JavaScript object properties
+  dynamically by using the bracket notation `[]`
+- [`bs.scope`](todo-fix-me.md): reach to deeper properties inside a JavaScript
+  object
+- [`bs.val`](#bind-to-global-javascript-functions): bind to global JavaScript
+  functions
+- [`bs.variadic`](#variadic-function-arguments): bind to a function taking
+  variadic arguments from an array
+
+Used to annotate arguments in `external` definitions:
+
+- [`bs`](#binding-to-callbacks): define function arguments as uncurried (manual)
+- [`bs.int`](#using-polymorphic-variants-to-bind-to-enums-and-string-types):
+  compile function argument to an int
+- [`bs.string`](#using-polymorphic-variants-to-bind-to-enums-and-string-types):
+  compile function argument to a string
+- [`bs.this`](#modeling-this-based-callbacks): bind to `this` based callbacks
+- [`bs.uncurry`](#binding-to-callbacks): define function arguments as uncurried
+  (automated)
+- [`bs.unwrap`](#approach-2-polymorphic-variant-bsunwrap): unwrap variant values
+
+Used in other places like records, fields, parameters, functions...:
+
+- `bs.as`: redefine the name generated in the JavaScript output code. Used in
+  [constant function arguments](#constant-values-as-arguments),
+  [variants](todo-fix-me.md), [polymorphic
+  variants](#using-polymorphic-variants-to-bind-to-enums-and-string-types) and
+  [record fields](#objects-with-static-shape-record-like).
+- [`bs.deriving`](todo-fix-me.md): generate getters and setters for some types
+- [`bs.inline`](todo-fix-me.md): forcefully inline constant values
+- [`bs.optional`](todo-fix-me.md): omit fields in a record (combines with
+  `bs.deriving`)
+
+**Extensions:**
+
+In order to use any of these extensions, you will have to add the melange PPX
+preprocessor to your project. To do so, add the following to the `dune` file:
+
+```bash
+(library
+ (name lib)
+ (modes melange)
+ (preprocess
+   (pps melange.ppx)))
+```
+
+The same field `preprocess` can be added to `melange.emit`.
+
+Here is the list of all extensions supported by Melange:
+
+- `bs.debugger`: insert `debugger` statements
+- `bs.external`: read global values
+- `bs.obj`: create JavaScript object literals
+- `bs.raw`: write raw JavaScript code
+- `bs.re`: insert regular expressions
+
+### Intro to `external`
```

Comment:
should we call this section "foreign function interface" or "modeling foreign JavaScript code"?

Reply:
Agree. Renamed in https://github.com/melange-re/documentation-site/pull/12/commits/d3da08456c9e3cdddb02a84e1348633f764e342f.